app/dbOperations/problems.go

```diff
@@ -0,0 +1,154 @@
+package dbOperations
+
+import (
+	"errors"
+	"fmt"
+
+	gremcos "github.com/supplyon/gremcos"
+	"github.com/supplyon/gremcos/api"
+	"github.com/supplyon/gremcos/interfaces"
+
+	modelStorage "github.com/imeplusplus/dont-panic-api/app/modelStorage"
+)
+
+func CreateProblem(cosmos gremcos.Cosmos, problem modelStorage.Problem) (modelStorage.Problem, error) {
+	_, err := GetProblemByName(cosmos, problem.Name)
+
+	if err == nil {
+		return modelStorage.Problem{}, fmt.Errorf("there is already a problem with name %v", problem.Name)
```

Comment:
minor: consider using typed errors, as they might make your functions easier to test and to reuse the error for other use cases:

```myerror.go
type MyAwesomeError struct {
    SomeField string
}

func (e MyAwesomeError) Error() string {
    return fmt.Errorf("some error occurred and the field is %s", e.SomeField)
}

func (e MyAwesomeError) Is(err error) bool {
    _, ok := err.(MyAwesomeError)
    return ok
}
```

Reply:
I added the tips to our notes so we can learn about them and replace in the code =]